src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/adapters.kt

```diff
@@ -0,0 +1,32 @@
+package uk.gov.justice.digital.hmpps.incentivesapi.service
+
+import uk.gov.justice.digital.hmpps.incentivesapi.dto.IepDetail
+import uk.gov.justice.digital.hmpps.incentivesapi.dto.prisonapi.IepLevel
+import uk.gov.justice.digital.hmpps.incentivesapi.jpa.NextReviewDate
+import uk.gov.justice.digital.hmpps.incentivesapi.jpa.PrisonerIepLevel
+import java.time.LocalDate
+
+fun PrisonerIepLevel.translate(incentiveLevels: Map<String, IepLevel>) =
+  IepDetail(
+    id = id,
+    bookingId = bookingId,
+    iepDate = reviewTime.toLocalDate(),
+    iepTime = reviewTime,
+    agencyId = prisonId,
+    iepLevel = incentiveLevels[iepCode]?.iepDescription ?: "Unmapped",
+    iepCode = iepCode,
+    reviewType = reviewType,
+    comments = commentText,
+    userId = reviewedBy,
+    locationId = locationId,
+    prisonerNumber = prisonerNumber,
+    auditModuleName = "INCENTIVES_API",
+  )
+
+fun List<PrisonerIepLevel>.toIepDetails(iepLevels: Map<String, IepLevel>): List<IepDetail> {
+  return this.map { review -> review.translate(iepLevels) }
+}
+
+fun List<NextReviewDate>.toMap(): Map<Long, LocalDate> {
```

Comment:
Could name this `toMapByBookingId` so folks know the shape of the collection returned?

Reply:
Could do, I'm also gonna rename `.translate()` to `toIepDetail()` as it still puzzles me when I look at it for a similar reason then.

Reply:
Done here: 5bad1d239e9e7b216bcd3eac016a3d33fab75420